examples/05-file-IO/02-hdf5_serialize_and_read.py

```diff
@@ -0,0 +1,154 @@
+# noqa: D400
```

Comment:
Hi @JennaPaikowsky, would it be possible that you review this example comments ? 
Just that you know, this PR is still draft because the code needs to be updated (I have a fix to complete on C++ side before updating). I plan on completing once the C++ fix is merged.